app/views/unattended/provisioning_templates/snippet/disk_enc_clevis_tang.erb

```diff
@@ -27,52 +27,66 @@ description: |
 <% if (@host.operatingsystem.family == 'Redhat' || @host.operatingsystem.name == 'Ubuntu') && tang_server_list.present? -%>
 
 cat > /tmp/rootdir-luks-device.sh << "EOF"
-#!/bin/sh
+#!/bin/bash
 #
-# Author Jan Löser <loeser@atix.de>
-# Published under the GNU Public Licence 3
+# SPDX-FileCopyrightText: 2024 Jan Löser <loeser@atix.de>
+# SPDX-FileCopyrightText: 2024 Martin Spiessl <spiessl@atix.de>
+# SPDX-License-Identifier: GPL-3.0-or-later
 #
-# This scripts tries to find the 1st LUKS device for / (root directory).
+# This script tries to find all LUKS devices mounted in /etc/crypttab.
 #
 set -o pipefail
 
-rootdev=$(df / --output=source | tail -n1)
-targetdev=$(readlink -f $rootdev)
-slavedev=$targetdev
-
-while : ; do
-    /sbin/cryptsetup luksDump $slavedev &>/dev/null && echo $slavedev && exit 0
+mapfile -t maybe_luks_devices < <(
+    awk '{print $1}' /etc/crypttab | \
+    xargs -rn1 find /dev -name | \
+    xargs -rn1 readlink -f
+)
+luks_devices=()
+while [[ ${#maybe_luks_devices[@]} -gt 0 ]] ; do
+    # classify maybe_luks_devices into luks_devices and non_luks_devices
+    non_luks_devices=()
+    for dev in "${maybe_luks_devices[@]}" ; do
+        /sbin/cryptsetup luksDump "$dev" &>/dev/null && luks_devices+=("$dev") || non_luks_devices+=("$dev")
+    done
+    # resolve non_luks_devices to discover new maybe_luks_devices
     set -e
-    slave=$(find /sys/class/block/$(basename $slavedev)/slaves -type l | head -n1)
-    slavedev=$(find /dev -name "$(basename $slave)" | head -n1)
+    maybe_luks_devices=()
+    for dev in "${non_luks_devices[@]}" ; do
+        mapfile -t  slaves < <(find "/sys/class/block/$(basename "$dev")/slaves" -type l)
+        for slave in "${slaves[@]}" ; do
+            slavedev=$(find /dev -name "$(basename "$slave")" | head -n1)
+            maybe_luks_devices+=("$slavedev")
+        done
+    done
     set +e
 done
-
-exit 1
+printf "%s\n" "${luks_devices[@]}"
 EOF
 
 # needs bash here because Ubuntu's sh (dash) doesn't support `-o pipefail` option
-luksdev=$(bash /tmp/rootdir-luks-device.sh)
-
-if [[ -n "$luksdev" ]]; then
-  echo "LUKS device found for '/': $luksdev"
+luksdevs=$(bash /tmp/rootdir-luks-device.sh)
+if [ -n "$luksdevs" ]; then
+  echo "LUKS devices found for '/': $luksdevs"
 
 <% if @host.operatingsystem.family == 'Redhat' -%>
   $PKG_MANAGER_INSTALL <%= packages_redhat %>
 <% elsif @host.operatingsystem.name == 'Ubuntu' -%>
   $PKG_MANAGER_INSTALL <%= packages_ubuntu %>
 <% end -%>
 
+for luksdev in $luksdevs; do
```

Comment:
if you add a for, the indentation is broken. 
the for need to have the same spaces as the "echo" in line 70, right?

Reply:
Indentation is pain in the ass with template tags.

I prefer to decouple the template tag indentation from the actual script indentation like we do already in the rest of the template (see line 72, 74, 76, ...). These if and loop conditions do not actually produce code (= bash statements) in the rendered script.

Means, let's keep it as it is, the rendered script's indentation looks good.